src/Core/Security/GeneralConfiguration.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Security;
+
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+
+/**
+ * Responsible for saving configuration options for security
+ */
+final class GeneralConfiguration implements DataConfigurationInterface
```

Comment:
We already have a few `GeneralConfiguration` classes existing, even if they are in different namespaces it's kind of a mess when searching for one so if we can avoid adding another one it would be nice

Besides, since it's the only configuration class for this domain I'm not sure the "general" distinction is relevant. Can we simply name this one `SecurityConfiguration` ?

Reply:
If you look at  `GeneralType`, we have 5 classes with that name.
For `GenerationConfiguration` whe have 3 classes with that name too :sweat_smile: 


We shouldn't be afraid of class name because of namespace and context.

In the future, more settings and forms will be added to this part. We have to be consistent with other forms.

Symfony also don't care about class name (regarding the namespace)
![image](https://user-images.githubusercontent.com/1462701/108869555-99a48500-75f7-11eb-84c5-45b125ae8010.png)
![image](https://user-images.githubusercontent.com/1462701/108869771-c9ec2380-75f7-11eb-95d7-694f80386043.png)

Reply:
Yes, that was my point too many classes with identical names (that's my problem 😅 ) I know we have the namespace but when you look for a class in your IDE, or a filename or even just the tab name when editing it's a bit confusing But that's just my opinion, I prefer when class names are self explanatory and generic names tend to hide this

Besides regarding Symfony your argument is only half true because these classes are part of different components which are all completely independent, this is not our case with a monolithic project 🙂

But I won't insist on this anyway, especially if I'm the only one who is bothered by this

Reply:
> know we have the namespace but when you look for a class in your IDE, or a filename or even just the tab name when editing it's a bit confusing

About this, I'm pretty sure you can configure it in your IDE (I can do it in emacs, so PHPStorm must do it :trollface: )

Reply:
Well for the tab it does display the last sub namespace But still it's not easy for research 😛

Reply:
I agree with @jolelievre . So we're two.

Sorry @PierreRambaud most people use VSCode or PHPStorm not emacs

Reply:
Feel free to update it I'm tired of this PR